### ISSUE TITLE
Random seed with float values fails on Pharo 12+

### ISSUE THREAD

**Issue by olekscode** [guillep/pharo-core#16684]
This works on all Pharo images:
```st
Random seed: 42.
Random seed: 0.
```
But this only works on Pharo 9, 10, 11:
```st
Random seed: 42.0.
Random seed: 0.5.
```
On Pharo 12 and 13 it throws an `Error: Improper store into indexable object`

**Comment by Ducasse** [comment]
Thanks for the report. We should add a test!

**Comment by olekscode** [comment]
This happens because the new implementation of `Random` in Pharo 12 uses `DoubleWordArray` to store the seed. The error comes from `DoubleWordArray >> at: put:` and the comment of this method says:
```
... fail if the value is not of the right type for this kind of collection
```
The "right type" is explained in the comment of `DoubleWordArray` class:
```
DoubleWordArrays store 64-bit unsigned Integer values.
```
So to solve this issue, we must either stop using DoubleWordArray in the implementation of Random or agree that the random seed myst always be an unsigned integer and write it in the comment of the `Random >> seed:` method

**Comment by tinchodias** [comment]
AFAIU this was fixed May 31 by #16738

Please reopen if not true
